bot.js

```diff
@@ -85,61 +85,57 @@ mongoClient.connect(err => {
     if (onlineStatus.length > 0) {
-        twitchDb.collection('pingRoles').find({}).toArray().then(pingRoles => {
-            if (pingRoles.length > 0) {
-                let twitchGetRequest = `https://api.twitch.tv/helix/streams?user_login=${onlineStatus[0].name}`;
-                for (let i = 1; i < onlineStatus.length; i++) {
-                    twitchGetRequest += '&user_login=' + onlineStatus[i].name;
-                }
-                axios.get(twitchGetRequest, {
-                    headers: {'client-id': config.twitchClientID, Authorization: 'Bearer ' + twitchAccessToken}
-                }).then(response => {
-                    console.log(Date());
-                    console.log(response.data.data);
-                    let foundArr = [];
-                    response.data.data.forEach(entry => {
-                        const foundIndex = onlineStatus.findIndex(element => element.name == entry.user_login);
-                        if (foundIndex >= 0 && onlineStatus[foundIndex].status != entry.type) {
-                            foundArr.push(entry.user_login);
-                            onlineStatus[foundIndex].status = entry.type;
-                            twitchEmbed.execute(client, entry, pingRoles, onlineStatus[foundIndex].channelIDs);
-                        } else if (foundIndex >= 0 && onlineStatus[foundIndex].status == entry.type) {
-                            foundArr.push(entry.user_login);
-                        } else if (foundIndex == -1) {
-                            console.log("Stream data received for non followed stream");
-                        }
-                    });
-                    onlineStatus.forEach((element, index) => {
-                        if (!foundArr.includes(element.name))
-                            onlineStatus[index].status = "offline";
-                    });
-                }).catch(err => {
-                    console.log(err.response.data);
-
-                    if (err.response.status == 401) {
-                        getTwitchAccessToken();
-                        setTimeout(checkTwitchChannels, 3000);
+        const pingRoles = await twitchDb.collection('pingRoles').find({}).toArray()
+        if (pingRoles.length > 0) {
+            let twitchGetRequest = `https://api.twitch.tv/helix/streams?user_login=${onlineStatus[0].name}`;
+            for (let i = 1; i < onlineStatus.length; i++) {
+                twitchGetRequest += '&user_login=' + onlineStatus[i].name;
+            }
+            try {
+                const response = await axios.get(twitchGetRequest, {
+                    headers: { 'client-id': config.twitchClientID, Authorization: 'Bearer ' + twitchAccessToken }
+                })
+                console.log(Date(), response.data.data);
+                let foundArr = [];
+                response.data.data.forEach(entry => {
+                    const foundElement = onlineStatus.find(element => element.name == entry.user_login)
```

Comment:
it looked like you were always using the found element, so i swapped findIndex for just `find`